copier/user_data.py

```diff
@@ -357,14 +363,19 @@ def get_multiline(self) -> bool:
         multiline = cast_answer_type(multiline, cast_str_to_bool)
         return bool(multiline)
 
-    def validate_answer(self, answer) -> bool:
+    def validate_answer(self, answer) -> Union[bool, str]:
         """Validate user answer."""
         cast_fn = self.get_cast_fn()
         try:
             cast_fn(answer)
-            return True
         except Exception:
             return False
+        else:
+            if self.validation is None:
+                return True
+
+        validation = self.render_validation(self.validation, answer)
+        return validation == '' or validation
```

Comment:
I think this is better, if we just add the line commented below to `render_value` (see following comments):

```suggestion
        try:
            validation = self.render_value(self.validation).strip()
        except UserMessageError as error:
            raise ValidationError(str(error)) from UserMessageError
        if validation:
            raise ValidationError(validation)
        return True
```